### PR TITLE
Clarify rust-analyzer binary  install

### PR DESCRIPTION
**Pull request by zoechi**

### CODE REVIEW

File: docs/user/readme.adoc [hunk 1]
```diff
@@ -94,7 +94,8 @@ $ git clone https://github.com/rust-analyzer/rust-analyzer.git && cd rust-analyz
 $ cargo xtask install
 ----
 
-You'll need Cargo, nodejs and npm for this.
+You'll need Cargo, xtask, nodejs and npm for this.
+Cargo-xtask can be found at https://github.com/matklad/cargo-xtask
```
Comment (kiljacken) [inline docs/user/readme.adoc:98]: The xtask binary is part of the rust-analyzer repository. It'll automatically build when calling `cargo xtask` from inside the rust-analyzer folder.

File: docs/user/readme.adoc [hunk 2]
```diff
@@ -108,18 +109,29 @@ Here are some useful self-diagnostic commands:
 * To log all LSP requests, add `"rust-analyzer.trace.server": "verbose"` to the settings and look for `Server Trace` in the panel.
 * To enable client-side logging, add `"rust-analyzer.trace.extension": true` to the settings and open the `Console` tab of VS Code developer tools.
 
-=== Language Server Binary
+=== rust-analyzer Language Server Binary
 
 Other editors generally require the `rust-analyzer` binary to be in `$PATH`.
 You can download the pre-built binary from the https://github.com/rust-analyzer/rust-analyzer/releases[releases] page. Typically, you then need to rename the binary for your platform, e.g. `rust-analyzer-mac` if you're on Mac OS, to `rust-analzyer` and make it executable in addition to moving it into a directory in your `$PATH`.
 
+On Linux to install the `rust-analyzer` binary into `~/.cargo/bin` which usually is already added to `$PATH`, this commands could be used
```
Comment (flodiebold) [inline docs/user/readme.adoc:117]: I don't think putting binaries not installed by cargo into `~/.cargo/bin` is a good idea.

Comment (zoechi) [reply docs/user/readme.adoc:117]: What fo you think is a better place?

Comment (bjorn3) [reply docs/user/readme.adoc:117]: Maybe `~/.local/bin`? There are a lot of programs that install themselves there. It isn't in `PATH` by default though.

File: docs/user/readme.adoc (hunk 1 above)
Comment (zoechi) [reply docs/user/readme.adoc]: Thanks for pointing that out. I changed it.

File: docs/user/readme.adoc (hunk 2 above)
Comment (zoechi) [reply docs/user/readme.adoc]: Thanks. I also think this is a better suggestion. I changed it.

File: docs/user/readme.adoc [hunk 3]
```diff
@@ -113,14 +113,16 @@ Here are some useful self-diagnostic commands:
...
 
 [source,bash]
 ----
-$ curl -L https://github.com/rust-analyzer/rust-analyzer/releases/latest/download/rust-analyzer-linux -o ~/.cargo/bin/rust-analyzer
-$ chmod +x ~/.cargo/bin/rust-analyzer
+$ curl -L https://github.com/rust-analyzer/rust-analyzer/releases/latest/download/rust-analyzer-linux -o ~/.local/bin/rust-analyzer
+$ chmod +x ~/.local/bin/rust-analyzer
 ----
 
+Ensure `~/cargo/bin` is listed in the `$PATH` variable.
```
Comment (bjorn3) [inline docs/user/readme.adoc:124]:
```suggestion
Ensure `~/local/bin` is listed in the `$PATH` variable.
```

Comment (lnicola) [reply docs/user/readme.adoc:124]: .local

Comment (zoechi) [reply docs/user/readme.adoc]: Done

**Comment by matklad** [comment]
Thanks @zoechi !

bors r+

**Pull request merged**